### PR TITLE
[DO NOT MERGE] First Release PR

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/callingcards/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/callingcards _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,72 +1,119 @@
...
+   - Any one of the aligners listed above in 'Prepare the Genome'
+   - Alignment QC ([`samtools`](http://www.htslib.org/),
+     [picard](https://broadinstitute.github.io/picard/),[rseqc](http://rseqc.sourceforge.net/))))
+4. Count Hops ([`callingCardsTools`](https://cmatkhan.github.io/callingCardsTools/))
+5. Present QC data ([`MultiQC`](http://multiqc.info/))
 
-<!-- TODO nf-core: Add full-sized test dataset and amend the paragraph below if applicable -->
+## Usage
 
-On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources.The results obtained from the full-sized test can be viewed on the [nf-core website](https://nf-co.re/callingcards/results).
```
Comment (FriederikeHanssen) [inline README.md:23]: same here

File: assets/input_schema.json [hunk 2]
```diff
@@ -14,23 +14,33 @@
             },
```
Comment (FriederikeHanssen) [inline assets/input_schema.json:14]: why are you renaming the file?

File: conf/modules.config [hunk 3]
```diff
@@ -38,4 +35,352 @@ process {
...
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+    // bwamem2_index can take a lot of memory for larger genomes. This gives
+    // the pipeline the chance to scale up to 120G for this. The enormous
+    // amount of memory should be fixed in versions of bwamem2 after ~2022 01
+    withName: BWAMEM2_INDEX {
+        maxRetries = 5
+        memory = { check_max( 24.GB * task.attempt, 'memory' ) }
+    }
```
Comment (FriederikeHanssen) [inline conf/modules.config:55]: this should go to the base config. All resource configurations are generally stored there to keep them in one place

File: main.nf [hunk 4]
```diff
@@ -16,15 +16,101 @@ nextflow.enable.dsl = 2
...
+    ' If the genome index is not equivalent to the main fasta file,' +
+    ' then omit the index and allow the pipeline to create it from' +
+    ' the concatenated fasta files.'
+}
+
+if (params.datatype == "mammals" && params.r1_bc_pattern == null){
+    exit 1, 'You have not specified a barcode pattern for mammalian data.' +
+    ' Please specify a barcode pattern using the `r1_bc_pattern` parameter.'
+}
+
```
Comment (FriederikeHanssen) [inline main.nf:113]: In my opinion any non template validation should go to callingcards.nf. In any case, can you replace `exit 1` with `error()` for better canceling

File: modules/local/bwa_aln_to_bam/main.nf [hunk 5]
```diff
@@ -0,0 +1,61 @@
+process BWA_ALN_TO_BAM {
```
Comment (FriederikeHanssen) [inline modules/local/bwa_aln_to_bam/main.nf:1]: why are you using the local module and the nf-core one? they look the same to me

File: modules/local/callingcardstools/mammals/concatQC/environment.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+channels:
```
Comment (FriederikeHanssen) [inline modules/local/callingcardstools/mammals/concatQC/environment.yml:1]: you sadly probably need to update all modules once more plus your local modules. They all need environment names or the pipeline won't work with conda :/ Maybe check the modules repo for an example

File: modules/local/callingcardstools/yeast/demultiplex/main.nf [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        -r1 ${reads[0]} \\
+        -r2 ${reads[1]} \\
+        -b ${barcode_details} \\
+        ${args}
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        callingcardstools: \$(echo \$(callingcardstools --version 2>&1) | sed 's/.* //')
+    END_VERSIONS
+
```
Comment (FriederikeHanssen) [inline modules/local/callingcardstools/yeast/demultiplex/main.nf:38]:
```suggestion
```

File: modules/local/concatFasta/main.nf [hunk 8]
```diff
@@ -0,0 +1,37 @@
+process CONCATFASTA {
+
+    conda "bioconda::coreutils=8.25"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    'https://depot.galaxyproject.org/singularity/coreutils:8.25--1' :
+    'biocontainers/coreutils:8.25--1' }"
+
+    input:
+        path(main_fasta)
+        path(additional_fasta)
```
Comment (FriederikeHanssen) [inline modules/local/concatFasta/main.nf:10]:
```suggestion
    path(main_fasta)
    path(additional_fasta)
```

File: modules/local/concatFasta/main.nf [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+    output:
+        path "*concat.fasta" , emit: fasta
+        path "versions.yml"  , emit: versions
```
Comment (FriederikeHanssen) [inline modules/local/concatFasta/main.nf:14]:
```suggestion
    path "*concat.fasta" , emit: fasta
    path "versions.yml"  , emit: versions
```

**Review by FriederikeHanssen — COMMENTED** [review]
The modules all look pretty nf-core like. I think it would be quite beneficial to add them in the long run to the nf-core/modules repo :) 


Hey! It looks super good already. Would it be possible to merge in the latest template update still?

I got about halfway through and will continue the review this week

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,14 +3,20 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## v0.0.0 - [date]
+## v1.0.0 - 2023-10-31
 
-Initial release of nf-core/callingcards, created with the [nf-core](https://nf-co.re/) template.
```
[the remark this answers is not in the document]
Comment (cmatKhan) [reply CHANGELOG.md:8]: I hadn't updated the template in quite some time. removing this was unintentional -- just a messy merge. ill add it back

File: README.md [hunk 11]
```diff
@@ -1,72 +1,119 @@
...
+organism type (either mammals or yeast) and either the name of the reference genome on `igenomes` or paths to the fasta/gff. It then parses the reads, counts the
+number of calling cards insertions, and provides some QC metrics.
 
-**nf-core/callingcards** is a bioinformatics best-practice analysis pipeline for A pipeline for processing calling cards data.
+![nf-core/rnaseq metro map](docs/images/callingcards_metro_diagram.png)
 
-The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The [Nextflow DSL2](https://www.nextflow.io/docs/latest/dsl2.html) implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies. Where possible, these processes have been submitted to and installed from [nf-core/modules](https://github.com/nf-core/modules) in order to make them available to all nf-core pipelines, and to everyone within the Nextflow community!
```
[the remark this answers is not in the document]
Comment (cmatKhan) [reply README.md:19]: I was worried about things like this. I hadn't updated the TEMPLATE for some time, and may have done a poor job merging when I did. I'll check again on the diffs between the current template and my current branch

File: main.nf (hunk 4 above)
Comment (cmatKhan) [reply main.nf:113]: there are two workflows, one more yeast, the other for mammals, so no single `callingcards.nf`. That itself is likely a point to consider -- a single `callingcards.nf` that calls different subworkflows depending on the `params.datatype`

File: README.md [hunk 12]
```diff
@@ -1,72 +1,119 @@
...
+1. Prepare the Genome
+   - optional masking with [bedtools](https://bedtools.readthedocs.io/en/latest/)
+   - Create aligner specific indicies. Available aligners:
+     - [bowtie](https://bowtie-bio.sourceforge.net/index.shtml)
```
[the remark this answers is not in the document]
Comment (cmatKhan) [reply README.md:25]: why remove this?

File: modules/local/bwa_aln_to_bam/main.nf (hunk 5 above)
Comment (cmatKhan) [reply modules/local/bwa_aln_to_bam/main.nf:1]: replaced with the fastq_align_bwa subworkflow

File: modules/local/concatFasta/main.nf (hunk 9 above)
Comment (cmatKhan) [reply modules/local/concatFasta/main.nf:14]: resolved in v1.1.0

File: modules/local/concatFasta/main.nf (hunk 8 above)
Comment (cmatKhan) [reply modules/local/concatFasta/main.nf:10]: resolved in v1.1.0

File: modules/local/callingcardstools/mammals/concatQC/environment.yml (hunk 6 above)
Comment (cmatKhan) [reply modules/local/callingcardstools/mammals/concatQC/environment.yml:1]: resolved in v1.1.0

File: README.md [hunk 13]
```diff
@@ -1,72 +1,119 @@
...
 
 ## Introduction
 
-<!-- TODO nf-core: Write a 1-2 sentence summary of what data the pipeline is for and what it does -->
+**nf-core/callingcards** is a bioinformatics pipeline that can be used to process raw Calling Cards data obtained from either mammals (human, mouse) or yeast. It takes a samplesheet, which describes the sample names, paths to the fastq files, and paths to the barcode details json files, a parameter which identifies the
+organism type (either mammals or yeast) and either the name of the reference genome on `igenomes` or paths to the fasta/gff. It then parses the reads, counts the
+number of calling cards insertions, and provides some QC metrics.
 
-**nf-core/callingcards** is a bioinformatics best-practice analysis pipeline for A pipeline for processing calling cards data.
+![nf-core/rnaseq metro map](docs/images/callingcards_metro_diagram.png)
```
[the remark this answers is not in the document]
Comment (cmatKhan) [reply README.md:20]: I attempted to fix this, and I think it is slightly improved.

but the diagram probably needs to be re-done. I was learning inkscape as i made this and distorted the text boxes while re-sizing.

**Pull request closed without merging**